### DISCUSSION TITLE
Fulfillment Inbound API v2024-03-20 Returning Empty Shipment

### DISCUSSION THREAD

**Discussion by belalali0626** [amzn/selling-partner-api-models#4425] [Q&A]
## Context

I am currently using the Amazon Selling Partner Fulfillment Inbound API v2024-03-20 to retrieve shipment data for past and current shipments, which I need to present to a client. After reading the documentation, I have made the following API calls:

1. `GET: /inbound/fba/2024-03-20/inboundPlans/`
   - This endpoint returns a list of Inbound Plans.

2. `GET: /inbound/fba/2024-03-20/inboundPlans/{Inbound Plan ID}/placementOptions`
   - Using the Inbound Plan ID obtained from the previous call, I make a request to this endpoint to retrieve placement options.

## Problem

The `placementOptions` endpoint is returning empty data, even though I expect the data to exist. I have seen discussions in other forums where users have successfully retrieved data using the now-deprecated `getShipments` endpoint from version 0, but I am unable to retrieve any data with the new endpoints.

I suspect that I might be misinterpreting the API flow or missing a crucial step in the process. I would greatly appreciate if someone with knowledge/expertise in these specific API endpoints could point out where I might be going wrong in my approach and confirm if I am following the correct flow.

**Comment by MetaMan2023** [comment]
Make sure you are calling "getInboundOperationStatus" after any POST call.

I made that mistake early on... all post calls take time and you must call getInboundOperationStatus to make sure the prior POST call completes.   It takes a while too, like 15 seconds.   Don't call it over and over super fast, or it may result in an error as well.   I have to wait like 4+ seconds between each request.

Once it returns a successful status you can make the next call in the sequence.

If you don't do this, I believe the prior call may not be completing so when you make the next call it's returning empty results because nothing is ready yet.

I believe the documentation does mention it but it's not very clear about this step -- there really should be a very clear example.

Also, in my testing, I believe it's kind of a "bug" in that it returns empty results rather than an error -- it should say "you have not completed the prior request" rather than returning empty results.

Hope this helps.

**Comment by MetaMan2023** [comment]
@belalali0626 

FYI I have a thread to track everyone's status on the new API transition and issues -- check it out if you get a chance.

https://github.com/amzn/selling-partner-api-models/discussions/4427
